CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKHitTester.java

```diff
@@ -52,17 +110,158 @@ public CVKHitTester(CVKVisualProcessor parent) {
+        CVKAssert(cvkImage == null);
+        CVKAssert(commandBuffer == null);
+    }
     
     
     // ========================> Swap chain <======================== \\    
     
     @Override
     protected int DestroySwapChainResources() { return VK_SUCCESS; }
     
+    private int CreateSwapChainResources() {
+        cvkVisualProcessor.VerifyInRenderThread();
+        CVKAssert(cvkSwapChain != null);
+        int ret = VK_SUCCESS;
+
+                
+        // We only need to recreate these resources if the number of images in 
+        // the swapchain changes or if this is the first call after the initial
+        // swapchain is created.
+        if (swapChainImageCountChanged) {
+            ret = CreateImages();
+            if (VkFailed(ret)) { return ret; }            
+
+            ret = CreateFrameBuffer();
+            if (VkFailed(ret)) { return ret; }
+
+             ret = CreateCommandBuffer();
+            if (VkFailed(ret)) { return ret; }
+            
+        } else {
+            // This is the resize path, image count is unchanged.
+        }
+        
+        swapChainResourcesDirty = false;
+        swapChainImageCountChanged = false;
+        
+        return ret;
+    } 
+    
+    
+    // ========================> Image <======================== \\
+    
+    private int CreateImages() {
+        CVKAssert(cvkSwapChain != null);
+        CVKAssert(cvkSwapChain.GetDepthFormat() != VK_FORMAT_UNDEFINED);
+        
+        int ret = VK_SUCCESS;
+   
+        try (MemoryStack stack = stackPush()) {
+            
+            int textureWidth = cvkSwapChain.GetWidth();
+            int textureHeight = cvkSwapChain.GetHeight();
+            int requiredLayers = 1;
+                        
+            // Create destination color image            
+            cvkImage = CVKImage.Create(cvkDevice, 
+                                            textureWidth, 
+                                            textureHeight, 
+                                            requiredLayers, 
+                                            //colorFormat, // Format TODO Not sure what the format should be - look at GL version
+                                            cvkSwapChain.GetColorFormat(),
+                                            VK_IMAGE_VIEW_TYPE_2D,
+                                            VK_IMAGE_TILING_LINEAR, // Tiling
+                                            VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT, // TODO Usage 
+                                            VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT , // TODO - properties?
+                                            VK_IMAGE_ASPECT_COLOR_BIT,
+                                            "CVKImage cvkImage");  // TODO - aspect mask
+            
+            if (cvkImage == null) {
+                return 1;
+            }
+            
+            // Create depth image 
+            cvkDepthImage = CVKImage.Create(cvkDevice, 
```

Comment:
We should be able to use the main depth image.  It doesn't need to be preserved across shaders.

Reply:
Hmm that's true, I'll add a todo in there to change after testing (not this commit)